### PR TITLE
Link to user profile

### LINKED ISSUE #7310: Link the username that added the project
As a follow-up to https://github.com/gitpod-io/gitpod/issues/5119, there should be a link to the Git provider profile of the username that added the project.

This was mentioned here[[1](https://github.com/gitpod-io/gitpod/pull/5128#discussion_r751438719)] and here[[2](https://github.com/gitpod-io/gitpod/pull/5128#discussion_r771569685_)] with a description: 

> Users with multiple integrations will be shown using the user-id of the first signup, which is not necessarily the user-id for the integration of the repo in question. E.g I used a different username for GitHub vs. GitLab and the "already added" message is showing my GitHub userid even though the repo is on GitLab.

### PR DESCRIPTION
**Pull request by laushinka**
## Description
Links the profile of the user who already added a project. 
The profile provider should match the repo provider. 
| UPDATED |
|-|
|Screenshot 2021-12-20 at 15 27 03|

## Related Issue(s)

Fixes #7310

## How to test
1. Have two different accounts using the same Git provider.
2. Account A create a group/org and add projects/repos, and invite account B as member with write access.
3. Account A adds project to Gitpod.
4. Account B tries to add the project and should be shown the username of account A. This username should link to account A's profile on the Git provider website.

## Release Notes

```release-note
Profile of the user who already added a project is linked.
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

Associated issue: *#7310*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by laushinka** [comment]
This is still a draft. @jldec Please help test using your [usecase](https://github.com/gitpod-io/gitpod/pull/5128#discussion_r771569685).

**Comment by laushinka** [comment]
Ready for that review and thumbs up! @jldec

**Comment by jldec** [comment]
@laushinka I think the code is not covering the specific case with 2 integrations each using a different user-id, which was mentioned in the issue:

> Users with multiple integrations will be shown using the user-id of the first signup, which is not necessarily the user-id for the integration of the repo in question. E.g I used a different username for GitHub vs. GitLab and the "already added" message is showing my GitHub userid even though the repo is on GitLab.

To reproduce:
1. create an account with 2 integrations (first GitHub then GitLab) using different user-ids
2. create a project with a GitLab repo
3. try to create a 2nd project on the same GitLab repo

The message will say

> [@user]() already
> added this repo

where @user is the **GitHub** user-id.
The link will go to a wrong (on non-existent) user on **GitLab**

The link should either go to the repo-owner's user-id (owner) on GitLab, or to the user-id on GitHub from integration 1.

File: components/dashboard/src/projects/NewProject.tsx
Comment (jldec) [inline components/dashboard/src/projects/NewProject.tsx:278]: nit: inconsistent `inUse?.userName` here, vs `inUse.userName` on line 331.

Comment (laushinka) [reply components/dashboard/src/projects/NewProject.tsx:278]: This is because in line 331, the condition already checks whether `inUse` exists, therefore the `?` check is no longer needed.

**Comment by laushinka** [comment]
> The link should either go to the repo-owner's user-id (owner) on GitLab, or to the user-id on GitHub from integration 1.

@jldec I managed to reproduce! Thanks for the steps.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

Associated issue: *#7310*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by laushinka** [comment]
@jldec Fixed the bug by checking that the owner provider matches the repo provider. Please test and confirm 🙏🏽

**Review by jldec — APPROVED** [review]
Thanks @laushinka - looks good.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 08afda6344f79eccaca4ed0305748a64f10a716e</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7312#pullrequestreview-837480936" title="Approved">jldec</a>*

Associated issue: *#7310*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jldec]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [jldec]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
